### PR TITLE
Rip out ZServer support.

### LINKED ISSUE #592: Remove ZServer support
- [x] from `setup.py`
- [x] from buildout.cfg
- [x] from code
- [x] from documentation

### PR DESCRIPTION
**Pull request by icemac**
Open questions/tasks:

- [x] Should zserver directives be removed from etc/example.conf? ← they already have been removed
- [x] Find out if `ZApplicationWrapper.__bobo_traverse__` is actually only called by ZServer, see https://github.com/zopefoundation/Zope/blob/b9a346903fe4a6874face0930876184ac321853e/src/App/ZApplication.py#L45-L47 ← yes, it is actually used by a ZServer based Zope but not by a WSGI based one.

fixes #592

### CODE REVIEW

File: src/ZPublisher/Publish.py [hunk 1]
```diff
@@ -14,28 +14,7 @@
...
-    publish_module_standard=(
-        'ZServer.ZPublisher.Publish:publish_module_standard'),
-    set_default_debug_mode=(
-        'ZServer.ZPublisher.Publish:set_default_debug_mode'),
-    set_default_authentication_realm=(
-        'ZServer.ZPublisher.Publish:set_default_authentication_realm'),
-)
-
-# BBB Zope 5.0
+# BBB Zope 6.0
```
Comment (dataflake) [inline src/ZPublisher/Publish.py]: Was this not ready for removal?

Comment (icemac) [reply src/ZPublisher/Publish.py]: You are right it can be removed in 5.0 but in another PR.

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2019-9-9 23:10 -0700:
> ...
>-# BBB Zope 5.0
>+# BBB Zope 6.0
>
>You are right it can be removed in 5.0 but in another PR.

Please keep "webdav" support (which apparently accidentally
was moved into the `ZServer` package).

**Comment by icemac** [comment]
Zope 4 using WSGI does not support WebDAV. There is #649 track re-enabling it. It would be a huge benefit if Zope 5 could support WebDAV again.
I personally neither have the need nor resources to re-implement WebDAV in Zope, so any help is welcome.

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2019-9-10 23:19 -0700:
>Zope 4 using WSGI does not support WebDAV. There is #649 track re-enabling it. It would be a huge benefit if Zope 5 could support WebDAV again.
>I personally neither have the need nor resources to re-implement WebDAV in Zope, so any help is welcome.

"WebDAV" used to be a top level package in Zope 2.
I assume that it was put below `ZServer` under the (wrong) assumption
it were dependent on `ZServer`.
Could you move it into a separate "zopefoundation/webdav"?

`OFS` and (I think) `ZPublisher` have special handling for `WebDAV`.
It would be quite difficult (complex monkey patching) to have
`WebDAV` without this special code.
In Zope 4, this was dependent on `ZServer` and it therefore risks
to get lost with the outphasing of `ZServer`.
The dependency should become one of `WebDAV`.

**Comment by dataflake** [comment]
I'm sure no one would complain if you just went ahead and did this yourself. 😉

**Comment by d-maurer** [comment]
Jens Vagelpohl wrote at 2019-9-11 03:12 -0700:
>I'm sure no one would complain if you just went ahead and did this yourself. ?

I do not yet know how I can move parts from one git repository
(including its history) to a different or a new one.

**Comment by dataflake** [comment]
There's a howto in the GitHub help pages: https://help.github.com/en/articles/splitting-a-subfolder-out-into-a-new-repository

**Comment by drfho** [comment]
Hi @icemac , @dataflake ,
today our Zope-latest testing system (working with Products.CMFCore) cannot start anymore, because some CMFCore-methods expect a variable OFS.bbb.HAS_ZSERVER. 
This comaptibility problem can quickly be solved by providing a reduced module OFS.bbb just for setting the variable:
[bbb_py.txt](https://github.com/zopefoundation/Zope/files/3600325/bbb_py.txt)
`HAS_ZSERVER = False`
Or do you prefer fixing Products.CMFCore?

**Comment by dataflake** [comment]
Zope `master` is the precursor of Zope 5. Since this is a major release backwards incompatibilities are OK. A lot of the removed code had already been emitting DeprecationWarnings so it is up to the packages that want to use the deprecated code to get fixed. Fix CMFCore.

**Comment by icemac** [comment]
@d-maurer I'd prefer to have `webdav` back here in the Zope repository. I am not sure if it could be  used standalone. As you said using it requires some changes in Zope which I do not want to have conditionally again.

Maybe only parts of 24d566b475ec60a5f872d39cdb63cb730fdc0f82 have to be reverted to get it back including its history.

**Comment by dataflake** [comment]
FYI: Instead of trying to fiddle with some git magic it is sufficient to simply copy the contents of the files in `src/webdav` from that commit diff (or in case they were updated in ZServer, from there) back into their old location. When you do a `git add` their history will be right back.

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2019-9-11 23:17 -0700:
>@d-maurer I'd prefer to have `webdav` back here in the Zope repository. I am not sure if it could be  used standalone. As you said using it requires some changes in Zope which I do not want to have conditionally again.

I have no objections against `webdav` directly in the Zope repository.

**Pull request merged**